paddle/fluid/pir/dialect/op_generator/ops_api_gen.py

```diff
@@ -142,6 +142,7 @@
     'uniform_random_batch_size_like',
     'c_reduce_min',
     'c_reduce_min_',
+    'pull_sparse_v2',
```

Comment:
```suggestion
    'push_sparse_v2',
```
下同。很多地方都是pull_sparse_v2，可以再确认下op签名是按照pull还是push翻译的

Reply:
另外从这里看
```
class PushSparseV2Op : public framework::OperatorWithKernel {
 public:
  using framework::OperatorWithKernel::OperatorWithKernel;

  void InferShape(framework::InferShapeContext* ctx) const override {}

 protected:
  phi::KernelKey GetExpectedKernelType(
      const framework::ExecutionContext& ctx) const override {
    return phi::KernelKey(OperatorWithKernel::IndicateVarDataType(
                              ctx, framework::GradVarName("Out")),
                          ctx.GetPlace());
  }
};
```
push_sparse_v2的infermeta为空，算子签名里面可以先不设置infermeta字段，应该不影响单测

Reply:
> 下同。很多地方都是pull_sparse_v2，可以再确认下op签名是按照pull还是push翻译的

我记得之前提到反向的问题，这两个算子里面 push应该是pull的反向，之前看得不够仔细，应该说错了。他们的attribute应该是一致的，输入输出的差别主要参考这里
```cpp
template <typename T>
class PushSparseV2OpMaker : public framework::SingleGradOpMaker<T> {
 public:
  using framework::SingleGradOpMaker<T>::SingleGradOpMaker;

 protected:
  void Apply(GradOpPtr<T> retv) const override {
    retv->SetType("push_sparse_v2");
    retv->SetInput("Ids", this->Input("Ids"));
    retv->SetInput(framework::GradVarName("Out"), this->OutputGrad("Out"));
    retv->SetInput("W", this->Input("W"));
    retv->SetOutput(framework::GradVarName("Out"), this->OutputGrad("Out"));
    retv->SetAttrMap(this->Attrs());
  }
};
```
这里set的才会出现在push的签名里，framework::GradVarName("Out") = “Out@GRAD”, this->OutputGrad("Out")不用关心，不属于Op签名的一部分，在旧IR下主要是设置 某个输入/输出 对应哪个变量。

Reply:
收到，抱歉

Reply:
意思是，ops 的 op : pull_sparse_v2, backward : push_sparse_v2 这样写吗？

Reply:
暂时先不考虑前反向，因为目前阶段主要是为了保证旧IR可以正确翻译到新IR上，所以这个点暂时不是特别重要。可以先这样注册算子：
```
- op : push_sparse_v2
  args : (Tensor[] ids, Tensor[] w, Tensor[] out_grad_in, int embeddingdim = 11, int tableid = 0, str accessorclass = "", str ctrlabelname = "", int paddingid = 0, bool scalesparsegrad = true, str[] inputnames = {}, bool is_distributed = true)
  output : Tensor[](out_grad_out){out_grad_in.size()}
  kernel :
    func : push_sparse_v2
    data_type : out_grad_in
  inplace: (out_grad_in -> out_grad_out)
```
先注册在`paddle/fluid/pir/dialect/operator/ir/ops.yaml`里吧，不同时注册它的反向了，push_sparse_v2也不需要infermeta，保持和旧IR逻辑一致即可。

但是这里为了兼容性保证，需要在opcompat.yaml里做相应修改，由于这里的修改只作用于翻译，所以我们不能修改yaml文件，而是要修改`paddle/fluid/ir_adaptor/translator/op_compat_gen.py`，增加如下语句(保持当前op_compat.yaml里的配置不变)：
```
op_arg_name_mappings['push_sparse_v2'].update({"out_grad_in":"Out@GRAD", "out_grad_out":"Out@GRAD"})
```

Reply:
意思是，pull_sparse_v2的都不要，都换成push_sparse_v2，InferMeta的也删除是吗？

Reply:
是的，因为pull_sparse_v2是另一个算子，这里它们两个的前反向关系并不重要，等后续需要的时候再去补充比较合适，毕竟现在没有合适的方式去测试。